api/models/Notification.js

```diff
@@ -89,11 +90,25 @@ module.exports = bookshelf.Model.extend({
         return this.sendJoinRequestPush()
       case 'approvedJoinRequest':
         return this.sendApprovedJoinRequestPush()
+      case 'announcement':
+        return this.sendPushAnnouncement()
       default:
         return Promise.resolve()
     }
   },
 
+  sendPushAnnouncement: function (version) {
```

Comment:
you could test this by using mockify* from test/setup/helpers to replace this.reader().sendPushNotification with a mock function and then see if it was called with the expected alertText and path

*doesn't look like there are any examples of using mockify with instance methods, but I don't see why it shouldn't work just as well as with class methods. If mockify doesn't work you should be able to replace it with a mock by hand so to speak.